snooty/semanticparser.py

```diff
@@ -9,12 +11,58 @@ def __init__(self) -> None:
 
     def run(self, pages: Dict[FileId, Page]) -> Dict[str, SerializableType]:
         # Specify which transformations should be included in semantic postprocessing
-        functions: List[
-            Callable[[Dict[FileId, Page]], Dict[str, SerializableType]]
-        ] = []
+        functions: List[Callable[[Dict[FileId, Page]], Dict[str, SerializableType]]] = [
+            self.slug_title
+        ]
         document: Dict[str, SerializableType] = {}
 
         for fn in functions:
             field: Dict[str, SerializableType] = fn(pages)
             document.update(field)
         return document
+
+    def slug_title(self, pages: Dict[FileId, Page]) -> Dict[str, SerializableType]:
+        # Function which returns a dictionary of slug-title mappings
+        slug_title_dict: Dict[str, SerializableType] = {}
+        for file_id in pages:
```

Comment:
[nit] Here's a more pythonic means of iterating over a dictionary:
```suggestion
        for file_id, page in pages.items():
```